go/def.bzl

```diff
@@ -90,6 +90,27 @@ def _relative_path_from_dir(from_dir, to_file):
     up = from_dir.count('/') + 1
   return up * "../" + to_file
 
+def build_file_tree(ctx, dest_dir, artifact_dict):
+  """build_file_tree creates the specified file structure.
+
+  Args:
+    dest_dir: The destination directory, a string.
```

Comment:
May want to comment that `dest_dir` should be a basename, not a path from the workspace root.